models/lwr/17x17_uo2_assembly/rodded/make_openmc_model.py

```diff
@@ -31,27 +32,25 @@
 #--------------------------------------------------------------------------------------------------------------------------#
 # Geometry definitions.
 assembly_bb.boundary_type = 'reflective'
-core_z_planes = [ openmc.ZPlane(z0=z) for z in np.linspace(0.0, geom.core_height, args.n_axial + 1) ]
-core_z_planes[0].boundary_type = 'reflective'
+fuel_center = openmc.ZPlane(z0 = 0.0, boundary_type = 'reflective')
+fuel_top = openmc.ZPlane(z0 = geom.core_height)
 
-all_cells = []
-for layer_idx, planes in enumerate(zip(core_z_planes[:-1], core_z_planes[1:])):
-  all_cells.append(openmc.Cell(name = f'UO2 Assembly Cell {layer_idx}', region = -assembly_bb & +planes[0] & -planes[1], fill = asmb['UO2_ROD']))
+asmb['UO2_ROD'].fill.pitch = (geom.pitch, geom.pitch, geom.core_height / args.n_axial)
+asmb['UO2_ROD'].fill.lower_left = (-pins_per_axis * geom.pitch / 2.0, -pins_per_axis * geom.pitch / 2.0, 0.0)
+asmb['UO2_ROD'].fill.universes = [ asmb['UO2_ROD'].fill.universes for i in range(args.n_axial) ]
+asmb['UO2_ROD'].region = asmb['UO2_ROD'].region & +fuel_center & -fuel_top
 
 refl_top = openmc.ZPlane(z0 = geom.core_height + geom.reflector_t, boundary_type = 'vacuum')
-all_cells.append(openmc.Cell(name='Axial Reflector Cell', fill = asmb['REF_ROD'], region=-assembly_bb & -refl_top & +core_z_planes[-1]))
+asmb['REF_ROD'].region = asmb['REF'].region & -refl_top & +fuel_top
 
 #--------------------------------------------------------------------------------------------------------------------------#
 # Setup the model.
-c5g7_model = openmc.Model(geometry = openmc.Geometry(openmc.Universe(cells = all_cells)), materials = openmc.Materials([mats['UO2'], mats['H2O'], mats['FISS'], mats['ZR_C'], mats['AL_C'], mats['BC4']]))
+c5g7_model = openmc.Model(geometry = openmc.Geometry(openmc.Universe(cells = [asmb['UO2_ROD'], asmb['REF_ROD']])), materials = openmc.Materials([mats['UO2'], mats['H2O'], mats['FISS'], mats['ZR_C'], mats['AL_C'], mats['BC4']]))
```

Comment:
Small thing to decrease verbosity here. Materials don't need to be provided to an openmc.Model object for CSG models. If they aren't present when the model is exported, the materials will be discovered on any material-filled cells.